.moai/specs/SPEC-UPDATE-REINSTALL-LOOP-002/acceptance.md

```
@@ -181,19 +183,41 @@ Expected: `--- PASS: TestUpdateDryRun_EmitsCleanReinstallPlan`. The captured out
+# (b) the guard passes — and its body asserts at RUNTIME that the seed reached permissions.deny
 go test ./internal/cli/ -run 'TestUpdateDryRun_ZeroMutation' -v 2>&1 | tail -5
 ```
 
 Expected: (a) prints **`≥1`**; (b) `--- PASS: TestUpdateDryRun_ZeroMutation`. The test hashes every file under the `t.TempDir()` project before and after the dry-run invocation and asserts the two maps are equal, including that no `.moai/backups/` directory was created.
 
 **Fixture requirement (a) is what makes this AC able to fail.** The fixture's `.claude/settings.json` MUST carry at least one of the twelve `retiredV2DenyEntries` literals (`internal/cli/update_deny_migration.go:20-33`) inside its `permissions.deny` array — for example `Write(~/.ssh/**)`. Without such an entry, `stripRetiredV2DenyEntries` (`internal/cli/update_deny_migration.go:42`) reaches `if removed == 0 { return nil }` and returns **before** its `os.WriteFile`, so an implementation that relocated the dry-run early return past `update.go:306-326` — the option `plan.md` §E M4 rejects — would still leave the tree byte-identical and the hash comparison would pass. The AC would report PASS on precisely the implementation it exists to reject.
 
-Observed early-return structure (four guard returns plus the `removed == 0` return, all ahead of the single write):
+**Command (a) is a text grep, and a text grep alone cannot carry this AC.** A prior draft ran the literal-match grep over the whole file, which a *comment* satisfies — grep cannot distinguish code from prose. Observed against a fixture whose `permissions.deny` array is empty and whose only occurrence of a literal is in a leading comment:
+
+```
```

markdownlint-cli2 (0.23.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)